### PR TITLE
ZEPHYR-291-hw-accelerated-crypto-driver

### PR DESCRIPTION
**Pull request by interfer**

### CODE REVIEW

File: tlsr9/CMakeLists.txt [hunk 1]
```diff
@@ -15,6 +15,17 @@ zephyr_include_directories(drivers/B91)
 if (NOT CONFIG_PM AND NOT CONFIG_BT_B91)
 zephyr_library_sources(drivers/B91/sys.c)
 endif() # no PM and no BLE external dependency
+
+# MbedTLS HW accelerated alternative
+
```
Comment (andriy-bilynskyy) [inline tlsr9/CMakeLists.txt]: remove space

File: tlsr9/CMakeLists.txt [hunk 2]
```diff
@@ -15,6 +15,17 @@ zephyr_include_directories(drivers/B91)
...
+if(CONFIG_MBEDTLS)
+add_subdirectory( crypto/mbedtls )
+endif() # CONFIG_MBEDTLS
+
+# ***
+
+# soc.c reference sources
+zephyr_library_sources_ifndef(CONFIG_BT_B91 drivers/B91/sys.c)
```
Comment (andriy-bilynskyy) [inline tlsr9/CMakeLists.txt]: why is this added?

File: tlsr9/crypto/mbedtls/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+# MbedTLS HW accelerated alternative
+
+zephyr_library_compile_definitions( MBEDTLS_CONFIG_FILE="${CONFIG_MBEDTLS_CFG_FILE}")
+
+zephyr_include_directories(
+    ./
+    ./internal/
+    ${ZEPHYR_MBEDTLS_MODULE_DIR}/include/
+    ${ZEPHYR_BASE}/modules/mbedtls/configs
```
Comment (andriy-bilynskyy) [inline tlsr9/crypto/mbedtls/CMakeLists.txt]: do we really need it?

File: tlsr9/crypto/mbedtls/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+# ***
+
+# ECP_ALT
+zephyr_library_sources(
+    ./internal/compatibility/ecp_alt.c
+    ./internal/compatibility/ecp_curves_alt.c
+    ./internal/ecp_alt_b91_backend_test.c
+    ./internal/ecp_alt_b91_backend.c
+    ./internal/test_utils.c
+    ./../../drivers/B91/pke.c
```
Comment (andriy-bilynskyy) [inline tlsr9/crypto/mbedtls/CMakeLists.txt]: it's handled by folder cmake

File: tlsr9/crypto/mbedtls/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    ./internal/entropy_poll_alt.c
+    ./../../drivers/B91/trng.c
```
Comment (andriy-bilynskyy) [inline tlsr9/crypto/mbedtls/CMakeLists.txt]: same as prev.

File: tlsr9/crypto/mbedtls/internal/entropy_poll_alt.c [hunk 6]
```diff
@@ -0,0 +1,59 @@
+/*
```
Comment (andriy-bilynskyy) [inline tlsr9/crypto/mbedtls/internal/entropy_poll_alt.c]: Seems we don't need this file

File: tlsr9/crypto/mbedtls/internal/multithread.c [hunk 7]
```diff
@@ -0,0 +1,53 @@
+/*
```
Comment (andriy-bilynskyy) [inline tlsr9/crypto/mbedtls/internal/multithread.c]: Why dymmy implementation is here?

File: tlsr9/crypto/mbedtls/internal/test_utils.c [hunk 8]
```diff
@@ -0,0 +1,51 @@
+/*
```
Comment (andriy-bilynskyy) [inline tlsr9/crypto/mbedtls/internal/test_utils.c]: test file shouldn't be here.

File: tlsr9/crypto/mbedtls/internal/test_utils.h [hunk 9]
```diff
@@ -0,0 +1,42 @@
+/**
```
Comment (andriy-bilynskyy) [inline tlsr9/crypto/mbedtls/internal/test_utils.h]: same as prev.

File: tlsr9/crypto/mbedtls/read_me.md [hunk 10]
```diff
@@ -0,0 +1,286 @@
+# MbedTLS HW acceleration for Telink B91 platform
```
Comment (andriy-bilynskyy) [inline tlsr9/crypto/mbedtls/read_me.md]: shouldn't be here

File: tlsr9/crypto/mbedtls/mbedtls-tlsr9-conf.h [hunk 11]
```diff
@@ -0,0 +1,511 @@
+/*
```
Comment (andriy-bilynskyy) [inline tlsr9/crypto/mbedtls/mbedtls-tlsr9-conf.h]: Remove commented lines and clean up the file

File: tlsr9/crypto/mbedtls/CMakeLists.txt (hunk 3 above)
Comment (interfer) [reply tlsr9/crypto/mbedtls/CMakeLists.txt]: no, we don't, fixed it

File: tlsr9/crypto/mbedtls/CMakeLists.txt (hunk 4 above)
Comment (interfer) [reply tlsr9/crypto/mbedtls/CMakeLists.txt]: it also depends on other options:

` # BLE reference sources
if (CONFIG_BT_B91)

zephyr_include_directories(
	ble
	ble/vendor/controller
)

zephyr_library_sources(
	drivers/B91/pke.c 
...
`

File: tlsr9/crypto/mbedtls/internal/multithread.c (hunk 7 above)
Comment (interfer) [reply tlsr9/crypto/mbedtls/internal/multithread.c]: fixed

File: tlsr9/crypto/mbedtls/internal/test_utils.c (hunk 8 above)
Comment (interfer) [reply tlsr9/crypto/mbedtls/internal/test_utils.c]: Could You please advice where to move them?

**Review by andriy-bilynskyy — APPROVED** [review]
